boot/bootchain.go

```diff
@@ -309,7 +311,23 @@ func readBootChains(path string) (pbc predictableBootChains, resealCount int, er
 	return wrapped.BootChains, wrapped.ResealCount, nil
 }
 
-func writeBootChains(pbc predictableBootChains, path string, resealCount int) error {
+func readRecoveryBootChains(path string) (pbc predictableBootChains, resealCount int, err error) {
+	inf, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, 0, nil
+		}
+		return nil, 0, fmt.Errorf("cannot open existing boot chains data file: %v", err)
+	}
+	defer inf.Close()
+	var wrapped predictableBootChainsWrapperForStorage
+	if err := json.NewDecoder(inf).Decode(&wrapped); err != nil {
+		return nil, 0, fmt.Errorf("cannot read boot chains data: %v", err)
+	}
```

Comment:
this could be a shared helper

Reply:
Yes, especially if we go with a separate file for the fallback chains.

Reply:
Using a separate file for recovery bootchains so all helpers can be used without duplication.